libs/partners/openai/langchain_openai/chat_models/base.py

```diff
@@ -706,21 +712,99 @@ def bind_tools(
         """
 
         formatted_tools = [convert_to_openai_tool(tool) for tool in tools]
-        if tool_choice is not None:
-            if isinstance(tool_choice, str) and (tool_choice not in ("auto", "none")):
-                tool_choice = {"type": "function", "function": {"name": tool_choice}}
-            if isinstance(tool_choice, dict) and (len(formatted_tools) != 1):
+        if tool_choice is not None and tool_choice:
+            if len(formatted_tools) != 1:
                 raise ValueError(
                     "When specifying `tool_choice`, you must provide exactly one "
                     f"tool. Received {len(formatted_tools)} tools."
                 )
-            if isinstance(tool_choice, dict) and (
-                formatted_tools[0]["function"]["name"]
-                != tool_choice["function"]["name"]
-            ):
+            if isinstance(tool_choice, str):
+                if tool_choice not in ("auto", "none"):
+                    tool_choice = {
+                        "type": "function",
+                        "function": {"name": tool_choice},
+                    }
+            elif isinstance(tool_choice, bool):
+                tool_choice = formatted_tools[0]
+            elif isinstance(tool_choice, dict):
+                if (
+                    formatted_tools[0]["function"]["name"]
+                    != tool_choice["function"]["name"]
+                ):
+                    raise ValueError(
+                        f"Tool choice {tool_choice} was specified, but the only "
+                        f"provided tool was {formatted_tools[0]['function']['name']}."
+                    )
+            else:
                 raise ValueError(
-                    f"Tool choice {tool_choice} was specified, but the only "
-                    f"provided tool was {formatted_tools[0]['function']['name']}."
+                    f"Unrecognized tool_choice type. Expected str, bool or dict. "
+                    f"Received: {tool_choice}"
                 )
             kwargs["tool_choice"] = tool_choice
         return super().bind(tools=formatted_tools, **kwargs)
+
+    @overload
+    def with_output_format(
+        self,
+        output_schema: Dict[str, Any],
+        *,
+        mode: Literal["tools", "json"] = "tools",
+        enforce_schema: bool = True,
+        return_single: bool = True,
+    ) -> Runnable[LanguageModelInput, Dict]:
+        ...
+
+    @overload
+    def with_output_format(
+        self,
+        output_schema: Type[_BM],
+        *,
+        mode: Literal["tools", "json"] = "tools",
+        enforce_schema: bool = True,
+        return_single: bool = True,
+    ) -> Runnable[LanguageModelInput, _BM]:
+        ...
+
+    def with_output_format(
+        self,
+        output_schema: Union[Dict[str, Any], Type[_BM]],
+        *,
+        mode: Literal["tools", "json"] = "tools",
+        enforce_schema: bool = True,
```

Comment:
enforce_schema sounds like validation, but this is not a validation step rather model forced to do tool invocation.

Reply:
what's a better name? want something not too openai tools specific if we're going to add this method (or similar methods) to other models